### PR TITLE
Antrea cni pack

### PR DESCRIPTION
**Pull request by ritawatson**
This PR documents the Antrea CNI pack. Please note the following: 

- I added a couple parameters to the table that seem to apply to this pack. 
- I made the first listed parameter "required" because my understanding is that users can't use the pack without enabling nodIPAM.
- Roshani, please review all sections of the doc for accuracy. 

[**Preview**](https://deploy-preview-1141--docs-spectrocloud.netlify.app/integrations/antrea-cni)
[**Jira**](https://spectrocloud.atlassian.net/browse/PAC-155)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1141#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | f2b4ee6516b2c0d8afa6d8d5760f47c635b1dfa3 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628121563230375936) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1141#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:20]:
```suggestion
Palette supports Antrea controller network interface (CNI) for Kubernetes clusters. Antrea CNI enables each pod to have exclusive IP addresses from the subnet with direct accessibility. 
```

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:22]:
```suggestion
Antrea leverages [Open vSwitch](https://www.openvswitch.org/) to implement pod networking and security features. Open vSwitch enables Antrea to implement Kubernetes network policies efficiently.
```

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:28]:
```suggestion
# Prerequisites
```

Because we only have one element, these have to get raised by one level. Once we add another major version and add a tab, we decrease all of these by one.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:45]:
```suggestion
# Parameters
```

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:60]:
```suggestion
# Usage
```

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:64]:
```suggestion
Antrea supports LoadBalancer services. Typically, implementing LoadBalancer services requires an external load balancer that is implemented by the Kubernetes cloud provider. 
```
the cloud provider feels odd as a proper noun here when it's used to reference all cloud providers.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:72]: There is also a docs page for using Antrea with MetalLB.  We should also incorporate that link, so maybe we should add these inline with the bullet?

https://antrea.io/docs/v1.9.0/docs/service-loadbalancer/#using-metallb-with-antrea

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:76]: This feels odd, we should prefix the scenario for where this is the answer.  See the [Spectro Proxy](https://docs.spectrocloud.com/integrations/frp#troubleshooting) for an example of how this should look.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-antrea-cni.md:91]:
```suggestion
 type = "helm"
```

**Review by karl-cardenas-coding — APPROVED** [review]
Besides the few suggestions, this PR looks great.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1141#pullrequestreview-1308200697" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1141#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:20]: I don't see a change here, but I do see one further down, which I committed.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md]: Done

File: content/docs/06-integrations/00-antrea-cni.md
Comment (roshanirathi) [inline content/docs/06-integrations/00-antrea-cni.md:39]: For palette, podCIDR mentioned in the k8s values.yaml will always take preference. If you want to use the Antrea CIDRs, (only podCIDR works, service CIDR doesn’t) you need to empty values for pod and service CIDR in the k8s pack.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (roshanirathi) [inline content/docs/06-integrations/00-antrea-cni.md:76]: IP address provided in k8s layer will always take preference. Refer to my previous comment

**Review by roshanirathi — CHANGES_REQUESTED** [review]

File: content/docs/06-integrations/00-antrea-cni.md
Comment (karl-cardenas-coding) [reply content/docs/06-integrations/00-antrea-cni.md:20]: @ritawatson you have an extra whitespace at the beginning of the second sentence.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:72]: Added link.

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | ff41fa258cd4cadbdca4a386c8295918ef0d8cab | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628469249011879936) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md]: fixed - thanks!

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 44bb2c21de73107cf52ddb204f1dc6cc695c53b2 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628470255321878528) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | eca482889607d14ad005596292ec808d947b60c1 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628471010367901696) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 8d1a0f71d379607745c63d7b973f5a7b8e576b7a | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628476295488737280) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 53f00bfa0aac5e51d4966c184a6cfe246db5f412 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1628485228777443328) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md]: @roshanirathi Please review the revised Troubleshooting section for accuracy.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md]: @roshanirathi I added VMware to this sentence: "Palette supports Antrea controller network (CNI) for VMware Kubernetes clusters." 

Is this accurate?

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:51]: @roshanirathi Please review Troubleshooting section. You can see the doc at the Preview link in the PR Description.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [inline content/docs/06-integrations/00-antrea-cni.md:20]: @roshanirathi 'VMware' is added in the first sentence - is it accurate to do so?

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [inline content/docs/06-integrations/00-antrea-cni.md:34]: @roshanirathi The prerequisite in line 34 is reworded. For consistency with 'Kubernetes manifest', 'Antrea controller' is changed here and throughout to say 'Antrea manifest'.

You can see the doc at the Preview link in the PR description.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [inline content/docs/06-integrations/00-antrea-cni.md:68]: @roshanirathi You mentioned ``serviceCIDR`` doesn't work. Should it be removed from the parameters table?

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [inline content/docs/06-integrations/00-antrea-cni.md:95]: @roshanirathi Troubleshooting section is updated. Please review for accuracy.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (roshanirathi) [inline content/docs/06-integrations/00-antrea-cni.md:101]: `kubeadm init` command is used when we are manually creating a k8s cluster. And if that is case then we won't have a kubernetes manifest to remove values from. 
Instead we can reword this to -
If you wish to use Antrea CIDRs and have deployed a cluster using Palette, ensure that you have done the following:

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:35]: Fixed.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:99]: Fixed

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md:101]: Fixed

**Comment by spectro-prow** [comment]
@ritawatson: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | fb9878d4a3b6fe421ac1fd3bbce9fba45e1930b1 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1141/librarium_build/1630284332058808320) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/06-integrations/00-antrea-cni.md
Comment (ritawatson) [reply content/docs/06-integrations/00-antrea-cni.md]: fixed.

File: content/docs/06-integrations/00-antrea-cni.md
Comment (roshanirathi) [inline content/docs/06-integrations/00-antrea-cni.md:51]: As discussed, this is not applicable in case of `kubeadm init`. This can be removed from here.

**Comment by karl-cardenas-coding** [comment]
Gitleaks handled in the `master` branch. These are false positives.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1141#pullrequestreview-1308200697" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1141#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
